doc/modules/cassandra/pages/architecture/cql-on-accord.adoc

```diff
@@ -0,0 +1,612 @@
+Accord.
+
+==== Detecting misrouted requests
+
+For Paxos this is resolved in the prepare phase where a failure to meet
+the required consistency level at the prepare phase means the operation
+does not run on Paxos. If the prepare phase is being performed to
+recover an existing transaction then it is allowed to proceed because
+recovery will deterministically create the same state every time it runs
+so it’s safe to repeat even after key or range migration has occurred
+since those would have already recovered the transaction.
+
+Accord determines an `executeAt` timestamp, that is deterministic even
+during transaction recovery, for each transaction that includes an epoch
+that corresponds to the epoch used by `TableMigrationState` and this is
+used to check all the tables and keys being touched in a transaction.
+`TxnQuery` then returns a retry on different system error if the any
+part of the transaction is not eligible to run on Accord.
+
+`ColumnFamilyStore` checks every `Mutation` to see if it is marked as
+allowing potential transaction conflicts. Paxos and Accord always mark
+their `Mutation`s as allowing potential transaction conflicts because
+they do the work to check for them directly, but non-SERIAL sources of
+`Mutation`s will be subject to that check and a
+`RetryOnDifferentSystemException` is thrown if the mutation is detected
+to be misrouted according to the latest cluster metadata available at
+the node attempting to apply that mutation.
+
+`ReadCommand` has a similar arrangement where each read command is
+marked with whether it allows potential transaction conflicts and when
+`executeLocally` is run the check is done against cluster metadata to
+determine whether or not to throw `RetryOnDifferentSystemException`.
+Accord always allows potential transaction conflicts on its read
+commands, but Paxos does not because Paxos does not need to read data in
+order to recover transactions.
+
+==== Splitting write requests
+
+For non-SERIAL writes the `Mutation` is split into the portion that will
+execute on Accord and the portion that will execute on Cassandra and the
+Accord portion is executed asynchronously while the Cassandra portion is
+executed synchronously. If either attempt fails due to misrouting the
+write is re-split with updated cluster metadata and retried without
+raising an error.
+
+Logged batches are currently always written to the system table and then
+split for execution, and if part of the batch fails then batchlog replay
+will replay the entire batch and re-split it in the process. Batchlog
+replay only makes a single attempt to replay before converting the batch
+contents to hints. If part of the batch was routed to Accord then there
+is no node to hint so there is a fake node that a hint is written to and
+when that hint is dispatched it will be split and then executed
+appropriately. In the future this needs to be simplified to writing the
```

Comment:
Is there a JIRA to track the future work if it needs to happen? Could also soften the language to `could be improved by`

Reply:
It's a correctness issue so this documentation will need to updated as part of https://issues.apache.org/jira/browse/CASSANDRA-20588